vespa-athenz/src/main/java/com/yahoo/vespa/athenz/client/ErrorHandler.java

```diff
@@ -0,0 +1,13 @@
+package com.yahoo.vespa.athenz.client;
+
+import com.yahoo.vespa.athenz.client.common.ClientBase;
+import org.apache.http.client.methods.HttpUriRequest;
+
+public interface ErrorHandler {
+    static ErrorHandler empty() {
+        return (r, e) -> {
+        };
+    }
+
+    void reportError(HttpUriRequest request, Exception error);
```

Comment:
Replace `HttpUriRequest` with a custom sub-interface containing the getters you need for metric reporting